### PR TITLE
fix(lib): allowing page size 1

### PR DESCRIPTION
**Pull request by StarlightIbuki**
KAG-5875

### CODE REVIEW

**Review by chobits — APPROVED** [review]

File: t/10-prefix.t [hunk 1]
```diff
@@ -328,27 +328,57 @@ key: test3 value: value3
...
-true
-true
-false.../lua-resty-lmdb/lua-resty-lmdb/lib/resty/lmdb/prefix.lua:34: 'page_size' can not be less than 2
+PAGE SIZE 1
+test value
+PAGE SIZE 2
+test value
+test1 value1
+PAGE SIZE 0
+page_size must be at least 1
```
Comment (chobits) [inline t/10-prefix.t:381]:
```suggestion
.*page_size must be at least 1
```

File: t/10-prefix.t [hunk 2]
```diff
@@ -328,27 +328,57 @@ key: test3 value: value3
...
+            res, more = p.page("test", "test", nil, 2)
+            assert(more)
+
+            assert(#res == 2)
+            for _, pair in ipairs(res) do
+                ngx.say(pair.key, " ", pair.value)
+            end
+
+            ngx.say("PAGE SIZE 0")
+            local ok, err= pcall(p.page, "test", "test", nil, 0)
```
Comment (chobits) [inline t/10-prefix.t:366]:
```suggestion
            local ok, err = pcall(p.page, "test", "test", nil, 0)
```

**Review by chobits — APPROVED** [review]

**Review by dndx — CHANGES_REQUESTED** [review]
`README.md` API doc for `page` need to be updated.

File: lib/resty/lmdb/prefix.lua [hunk 3]
```diff
@@ -31,10 +31,18 @@ function _M.page(start, prefix, db, page_size)
...
     end
 
-    assert(page_size >= 2, "'page_size' can not be less than 2")
+    assert(page_size >= 1, "page_size must be at least 1")
+
+    local query_size = page_size
+
+    -- the function ngx_lua_resty_lmdb_ffi_prefix requires at least 2 operations
+    -- special handling for the case when page_size is less than 2
+    if query_size < 2 then
```
[the remark this answers is not in the document]
Comment (dndx) [reply lib/resty/lmdb/prefix.lua:40]: Also `local math_max = math.max` as well.

**Review by chobits — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -234,8 +234,7 @@ from the `txn` table when `commit()` returned an error is undefined.
 Return all keys `>= start` and starts with `prefix`. If `db` is omitted,
 it defaults to `"_default"`.
 
-If `page_size` is specified, up to `page_size` results will be returned. However,
-`page_size` can not be set to less than `2` due to internal implementation limitations.
+If `page_size` is specified, up to `page_size` results will be returned.
```
Comment (dndx) [inline README.md:237]:
```suggestion
If `page_size` is specified, up to `page_size` results will be returned.
`page_size` can not be less than `1`.
```

**Review by dndx — CHANGES_REQUESTED** [review]

**Comment by dndx** [comment]
@StarlightIbuki This PR needs to fix it's tests and rebase with the correct `TEST` number in the `.t` file.

**Review by chronolaw — APPROVED** [review]

**Review by dndx — APPROVED** [review]

**Pull request merged**